pkg/agent/flowexporter/exporter/exporter_perf_test.go

```diff
@@ -0,0 +1,224 @@
+// +build !race
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"math/big"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vmware/go-ipfix/pkg/registry"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	"antrea.io/antrea/pkg/agent/flowexporter/connections"
+	"antrea.io/antrea/pkg/agent/flowexporter/flowrecords"
+)
+
+const (
+	testNumOfConns           = 100000
+	testNumOfDenyConns       = 100000
+	testNumOfDyingConns      = 10000
+	testNumOfInactiveRecords = 10000
+	testNumOfIdleDenyConns   = 10000
+	testBufferSize           = 1048
+)
+
+var exp *flowExporter
+
+var count = 0
+
+/*
+Sample output:
+	go test -test.v -run=BenchmarkExport -test.benchmem -bench=. -memprofile memprofile.out -cpuprofile profile.out
+	goos: linux
+	goarch: amd64
+	pkg: antrea.io/antrea/pkg/agent/flowexporter/exporter
+	cpu: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
+	BenchmarkExport
+		exporter_perf_test.go:80:
+			Summary:
+			Number of conntrack connections: 100000
+			Number of dying conntrack connections: 10000
+			Number of deny connections: 100000
+			Number of idle deny connections: 10000
+			Total connections received: 182861
+	BenchmarkExport-2   	       1	5469929206 ns/op	656425016 B/op	17396982 allocs/op
```

Comment:
Do you know why we are running the benchmark only one even after adding b.N in the test?
I see the same thing in flow aggregator tests. Not sure what is the reason. 
Should we move to regular test type and track the mem or allocation stats ourselves?
@antoninbas

Reply:
I found if reducing the number of connections, it will run multiple times. It may be dependent on capacity of the machine the test is running on. Here is part of the result:
```
    exporter_perf_test.go:61: 
        Summary:
        Number of conntrack connections: 10000
        Number of dying conntrack connections: 1000
        Number of deny connections: 10000
        Number of idle deny connections: 1000
        Total connections received: 168554
    exporter_perf_test.go:61: 
        Summary:
        Number of conntrack connections: 10000
        Number of dying conntrack connections: 1000
        Number of deny connections: 10000
        Number of idle deny connections: 1000
        Total connections received: 199035
BenchmarkExport-2   	     206	   7352172 ns/op	  492372 B/op	   17727 allocs/op
PASS
ok  	antrea.io/antrea/pkg/agent/flowexporter/exporter	7.627s
```

Reply:
Interesting. Did not know that it depends on the runtime and resource consumption of the test.

Reply:
It will stop running if one iteration takes too long.

I wonder if we should have different benchmarks for the different types of connections. Would it make sense?